jsonschema-hyperschema.xml

```diff
@@ -124,152 +129,235 @@
+                    <t>
+                        For a given link, an implementation MUST make the values of all
+                        target attribute keywords directly available to the client.
+                        Implementations MAY provide additional interfaces for using this
+                        information, which is discussed in each keyword's section.
+                    </t>
+                    <t>
+                        For a given link, an implementation MUST make the value of each
+                        input schema keyword directly available to the client.
+                    </t>
+                </section>
+            </section>
+            <section title="Additional functionality">
                 <t>
-                    If the subschema for a "contains" keyword contains hyper-schema keywords they
-                    MUST be applied to all array elements that validate against the schema.  While
-                    finding a single validating element is sufficient to determine the validation
-                    outcome, when hyper-schema keywords are present, the subschema MUST be evaluated
-                    against all array elements.
+                    Implementations MAY provide additional functionality coordinating the use of
+                    JSON Hyper-Schema with other specifications, including but not limited to
+                    JSON Schema validation, serializing payloads based on media types, and
+                    communicating with resources via protocols as indicated by URI schemes.
+                    <cref>
+                        This is vague.  I'm trying to convey that most uses of Hyper-Schema
+                        would benefit from including integration with the media types, URI schemes,
+                        and protocols, such as automatically validating submission data against
+                        "submissionSchema", encoding it according to "submissionMediaType",
+                        examining the URI for the "http" or "https" scheme, validating JSON
+                        representations of headers against "headerSchema", serializing them
+                        into HTTP header format, and constructing the appropriate POST request.
+                        While many of those activities are technically outside of the scope of
+                        Hyper-Schema, coordinating all of them is specific to Hyper-Schema and
+                        beneficial to automate.
+                    </cref>
                 </t>
             </section>
-        </section>
-
-        <section title="Meta-schema">
-            <t>
-                The current URI for the JSON Hyper-Schema meta-schema is
-                &lt;http://json-schema.org/draft-06/hyper-schema#&gt;.
-            </t>
-        </section>
-
-        <section title="Schema keywords">
-            <section title="base">
+            <section title="Streaming parsers">
                 <t>
```

Comment:
I don't get the point of this new "streaming parsers" section, can you explain?

Reply:
@dlax that comes in part from @awwright talking about writing a streaming parser.  When working out the testable requirements finding links, it occurred to me that it is not possible to find all links without processing the entirety of the instance document.

This is especially true for locating links based on context, because `anchor` and `anchorPointer` can adjust the context.  There's no way to tell, without searching the entire document, whether a link defined later on sets the current location as its context.

In the case of a streamed instance, the definition of "entire document" may be difficult to determine.  There may not even be a fixed-scope document, as the source of the stream may be adding to it based on other events.

So, if I wrote the requirement around find-by-context in terms of processing the entire document, then you could end up blocking indefinitely in a streaming parser situation.  I felt like that warranted an alternate consideration to provide some information while acknowledging that it is technically incomplete.

@awwright can comment on whether the section is accurate to what he had in mind, and whether it is really necessary.  But that's why it's there.